src/Dapr.Jobs/Extensions/Helpers/Serialization/DaprOneTimeJobsSerializationExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Text;
+using System.Text.Json;
+
+namespace Dapr.Jobs.Extensions.Helpers.Serialization;
+
+/// <summary>
+/// Provides helper extensions for performing serialization operations when scheduling one-time Cron jobs for the developer.
+/// </summary>
+public static class DaprOneTimeJobsSerializationExtensions
+{
+    /// <summary>
+    /// Schedules a one-time job.
+    /// </summary>
+    /// <param name="client">The <see cref="DaprJobsClient"/> instance.</param>
+    /// <param name="jobName">The name of the job being scheduled.</param>
+    /// <param name="scheduledTime">The point in time when the job should be run.</param>
+    /// <param name="payload">The main payload of the job expressed as a JSON-serializable object.</param>
+    /// <param name="jsonSerializerOptions">Optional options to use for the <see cref="JsonSerializer"/>.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
+    [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+    public static async Task ScheduleOneTimeJobWithPayloadAsync(this DaprJobsClient client, string jobName, DateTime scheduledTime,
+        object payload, JsonSerializerOptions? jsonSerializerOptions = null,
+        CancellationToken cancellationToken = default)
+    {
+        ArgumentNullException.ThrowIfNull(payload, nameof(payload));
+
+        var serializerOptions = jsonSerializerOptions ?? new JsonSerializerOptions(JsonSerializerDefaults.Web);
```

Comment:
The serialization/deserialization logic across the extension methods could probably be consolidated into common serialize/deserialize helper methods.

Reply:
They're close, but I don't know if they can necessarily be consolidated. One takes an object that can be presumably JSON serialized and encoded to UTF-8 bytes and the other takes what's presumably an already serialized string and encodes that to UTF-8 bytes without any further serialization. The idea for the former is that it keeps the SDK aligned with the other packages in that if you pass it serializable data, it'll just work, but the latter lets you handle serialization yourself.

I considered making the latter even just accept a byte[] so it would allow the user to handle serialization, compression and whatever else they wanted to do with the payload themselves and completely opt out of this package doing any of that lifting. Thoughts?

Reply:
I consolidated in-so-far as there's only a single `ScheduleJobsAsync` method on the client now, but there are still the two serializer extension methods for accepting a string -> UTF-8 bytes and object -> JSON serialization -> UTF-8 bytes (as opposed to the `ReadOnlyMemory<byte>` in on the `ScheduleJobsAsync` method itself which assumes the user took on serialization (and optionally compression) on their own.